frontend/public/components/nav/PinnedResource.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react';
+import { useDrag, useDrop } from 'react-dnd';
+import { useTranslation } from 'react-i18next';
+import * as classNames from 'classnames';
+import { Button } from '@patternfly/react-core';
+import { MinusCircleIcon, GripVerticalIcon } from '@patternfly/react-icons';
+import { K8sModel, modelFor, referenceForModel } from '../../module/k8s';
+import confirmNavUnpinModal from './confirmNavUnpinModal';
+import { NavLinkComponent, ResourceClusterLink, ResourceNSLink, RootNavLink } from './items';
+
+import './PinnedResource.scss';
+
+type PinnedResourceProps = {
+  resource?: string;
```

Comment:
to me, it looked confusing to have a prop of name `resource` of type `string`. Is it a resource reference? 

can we have a diff prop name?